src/condor_io/condor_auth_ssl.cpp

```diff
@@ -1023,14 +1023,30 @@ Condor_Auth_SSL::server_verify_scitoken()
 	std::string issuer, subject;
 	long long expiry;
 	std::vector<std::string> bounding_set;
+	std::vector<std::string> groups;
+	std::string jti;
 	if (!htcondor::validate_scitoken(m_client_scitoken, issuer, subject, expiry,
-		bounding_set, mySock_->getUniqueId(), err))
+		bounding_set, groups, jti, mySock_->getUniqueId(), err))
 	{
 		dprintf(D_SECURITY, "%s\n", err.getFullText().c_str());
 		return false;
 	}
+	classad::ClassAd ad;
+	if (!groups.empty()) {
+		std::stringstream ss;
+		bool first = true;
+		for (const auto &grp : groups) {
+			ss << (first ? "" : ",") << grp;
+			first = false;
+		}
+		ad.InsertAttr(ATTR_TOKEN_GROUPS, ss.str());
+	}
+	if (!jti.empty()) {
+		ad.InsertAttr(ATTR_TOKEN_ID, jti);
+	}
+	ad.InsertAttr(ATTR_TOKEN_ISSUER, issuer);
+	ad.InsertAttr(ATTR_TOKEN_SUBJECT, subject);
```

Comment:
What's the logic behind omitting only `ATTR_TOKEN_ID` if it's empty?

Reply:
The other two attributes are mandatory - it's an implementation bug if this is reachable and they aren't set.  This one is optional and up to the external issuer as to whether it exists.

Reply:
Sorry.  I missed that `htcondor::validate_scitoken()` will fail if issue or subject aren't set.